crates/utils/src/errors.rs

```diff
@@ -0,0 +1,14 @@
+#[derive(Debug)]
+pub enum ApiError {
+    ApiInitialisationFailed(String),
+    ApiServeFailed(String),
+    AddressResolutionFailed(String),
+    EndpointToSocketFailed(String),
+    DatabaseConnectionFailed(String),
+}
+
+impl std::fmt::Display for ApiError {
```

Comment:
Please use `thiserror` for `ApiError` and write display implementation using its macro. We use intensively `thiserror`, for example: https://github.com/0xPolygonMiden/miden-node/blob/next/block-producer/src/errors.rs#L16-L47